src/object.d

```diff
@@ -15,6 +15,7 @@ alias size_t = typeof(int.sizeof);
 alias ptrdiff_t = typeof(cast(void*)0 - cast(void*)0);
 
 alias sizediff_t = ptrdiff_t; // For backwards compatibility only.
+alias noreturn = typeof(*null);  /// bottom type
```

Comment:
Do we want it in `object.d` ? It's a rare enough case that I think we can live with `typeof(*null)` in the signature.

Reply:
We call it noreturn in the documentation, code, and error messages. Because of the way module scopes work, it shouldn't interfere with user use of the name, just like it doesn't for string.